OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -103,7 +105,7 @@ public void Damaged(Actor self, AttackInfo e)
 			}
 
 			// not a lot we can do about things we can't hurt... although maybe we should automatically run away?
-			if (!attack.HasAnyValidWeapons(Target.FromActor(attacker)))
+			if (!attackBases.Any(Exts.IsTraitEnabled) || !attackBases.Where(a => a.HasAnyValidWeapons(Target.FromActor(attacker))).Any(Exts.IsTraitEnabled))
```

Comment:
This simplifies to `!attackbases.Any(a => !a.IsTraitDisabled && a.HasAnyValidWeapons(Target.FromActor(attacker)))`